http.go

```diff
@@ -75,11 +78,19 @@ func (s *Server) ServeHTTPS() {
 	}
 
 	var err error
-	config.Certificates = make([]tls.Certificate, 1)
-	config.Certificates[0], err = tls.LoadX509KeyPair(s.Opts.TLSCertFile, s.Opts.TLSKeyFile)
+	servingCertProvider, err := dynamiccertificates.NewDynamicServingContentFromFiles("serving", s.Opts.TLSCertFile, s.Opts.TLSKeyFile)
 	if err != nil {
 		log.Fatalf("FATAL: loading tls config (%s, %s) failed - %s", s.Opts.TLSCertFile, s.Opts.TLSKeyFile, err)
 	}
+	go servingCertProvider.Run(1, context.TODO().Done())
+
+	config.GetCertificate = func(_ *tls.ClientHelloInfo) (*tls.Certificate, error) {
+		// this disregards information from ClientHello but we're not doing SNI anyway
+		cert, key := servingCertProvider.CurrentCertKeyContent()
```

Comment:
question: as this will be invoked with every TLS handshake, we are relying here on the underlying caching mechanism of the client implementation, correct?

Reply:
Which client caching mechanisms do you have in mind? Does https://github.com/openshift/oauth-proxy/pull/152#discussion_r378115167 answer your question?

Reply:
sounds, good the above comment answers it.